handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -235,37 +235,39 @@ protected void deallocate() {
         } finally {
             readerLock.unlock();
         }
-        ssl = SSL.newSSL(context.ctx, !context.isClient());
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
I didn't catch the duplicate call to `SSL.newSSL` ... the access to `context.ctx` must be protected via the reader lock. Please use my previous suggestion using a final variable to ensure no other threads see `ssl` before it is initialized here.